modules/core/src/main/kotlin/datamaintain/core/step/check/Checker.kt

```diff
@@ -0,0 +1,83 @@
+package datamaintain.core.step.check
+
+import datamaintain.core.Context
+import datamaintain.core.script.ScriptWithContent
+import datamaintain.core.step.check.rules.CheckRule
+import datamaintain.core.step.check.rules.ScriptType
+import datamaintain.core.step.check.rules.contracts.FullContextCheckRule
+import datamaintain.core.step.check.rules.contracts.ScriptCheckRule
+import datamaintain.core.step.check.rules.contracts.ScriptWithContextCheckRule
+import datamaintain.core.step.check.rules.implementations.ExecutedScriptsNotRemovedCheck
+import mu.KotlinLogging
+import java.lang.IllegalArgumentException
+
+private val logger = KotlinLogging.logger {}
+
+val allCheckRuleNames: Sequence<String> = sequenceOf(
+        ExecutedScriptsNotRemovedCheck.NAME
+)
+
+class Checker(private val context: Context) {
+    fun check(checkedData: CheckerData): List<ScriptWithContent> {
+        logger.info { "Check scripts..." }
+
+        // we want to ensure all rule can be built before to launch the first check,
+        // so end the sequence stream by building a list.
+        val rules = context.config.checkRules
+                .map { buildCheckRule(it) }
+                .toList()
+
+        // All rules exist so we can launch them
+        rules.forEach { executeRule(it, checkedData) }
+
+        logger.info { "All check rules passed !" }
```

Comment:
I would say "All check rules were executed!" because "passing" is mainly used for a test, which is not the case here.